pennylane/gradients/adjoint_metric_tensor.py

```diff
@@ -54,13 +60,28 @@ def _group_operations(tape):
     return trainable_operations, group_after_trainable_op
 
 
+def _expand_trainable_multipar(
+    tape: qml.tape.QuantumTape,
+) -> (Sequence[qml.tape.QuantumTape], Callable):
+    """Expand trainable multi-parameter operations in a quantum tape."""
+
+    interface = qml.math.get_interface(*tape.get_parameters())
+    use_tape_argnum = interface == "jax"
+    expand_fn = qml.transforms.create_expand_trainable_multipar(
+        tape, use_tape_argnum=use_tape_argnum
+    )
+    return [expand_fn(tape)], lambda x: x[0]
+
+
 @partial(
     transform,
+    expand_transform=_expand_trainable_multipar,
     classical_cotransform=_contract_metric_tensor_with_cjac,
     is_informative=True,
+    use_argnum_in_expand=True,
 )
 def adjoint_metric_tensor(
-    tape: qml.tape.QuantumTape,
+    tape: qml.tape.QuantumTape, argnums=None
```

Comment:
I don't think it is necessary here, because the qnode will use argnums expand them and set them on the tape directly. In group operation you can keep the old logic for everything that is not jax and set argnums=tape.trainable_params for Jax.

Reply:
See `test_adjoint_metric_tensor.py` for `test_correct_output_qnode_jax` with parameters `fubini_ansatz4` and its corresponding params `(np.array(-0.1735, requires_grad=True), np.array([-0.1735, -0.2846, -0.2846], requires_grad=True))`. In this test case, two parameters are passed to the qnode, and by default, only the first parameter is considered trainable, and this is due to `argnums = [0] if qnode.interface in ["jax", "jax-jit"] and argnums is None else argnums` this line in `transform_program.py`. So if there are more than one jax arrays that are all supposed to be trainable, we need to provide users with the option of specifying `argnums`.

Reply:
Look at the other transforms, you can specify `argnums` even when they are not argument of the function. Because apply the transforms just populates the transform program. The qnode uses the argnums but not the transform itself.

Reply:
Thanks! I just made the update.